src/CoreFoundation/CFProxySupport.cs

```
@@ -580,61 +580,105 @@ public static CFProxySettings GetSystemProxySettings ()
 			/* CFStringRef __nonnull */ IntPtr proxyAutoConfigurationScript,
 			/* CFURLRef __nonnull */ IntPtr targetURL,
-			/* CFProxyAutoConfigurationResultCallback __nonnull */ IntPtr cb,
-			/* CFStreamClientContext * __nonnull */ IntPtr clientContext);
+			/* CFProxyAutoConfigurationResultCallback __nonnull */ CFProxyAutoConfigurationResultCallback cb,
+			/* CFStreamClientContext * __nonnull */ ref CFStreamClientContext  clientContext);
 		
-		public static CFRunLoopSource ExecuteProxyAutoConfigurationScript (NSString proxyAutoConfigurationScript, NSUrl targetURL, CFProxyAutoConfigurationResultCallback resultCallback, CFStreamClientContext clientContext)
+		public static CFProxy[] ExecuteProxyAutoConfigurationScript (string proxyAutoConfigurationScript, Uri targetUrl)
 		{
 			if (proxyAutoConfigurationScript == null)
-				throw new ArgumentNullException ("proxyAutoConfigurationScript");
+				throw new ArgumentNullException (nameof (proxyAutoConfigurationScript));
 
-			if (targetURL == null)
-				throw new ArgumentNullException ("targetURL");
+			if (targetUrl == null)
+				throw new ArgumentNullException (nameof (targetUrl));
 
-			if (resultCallback == null)
-				throw new ArgumentNullException ("resultCallback");
+			using (var pacScript = new NSString (proxyAutoConfigurationScript)) 
+			using (var url = new NSUrl (targetUrl.AbsoluteUri)) {
+				if (url == null)
+					return null;
 
-			if (clientContext == null)
-				throw new ArgumentNullException ("clientContext");
+				CFProxy[] proxies = null;
+				var runLoop = CFRunLoop.Current;
 
-			IntPtr source = CFNetworkExecuteProxyAutoConfigurationScript (proxyAutoConfigurationScript.Handle, targetURL.Handle, resultCallback, clientContext);
-			return (source == IntPtr.Zero) ? null : new CFRunLoopSource (source);
+				CFProxyAutoConfigurationResultCallback cb = delegate (IntPtr client, IntPtr proxyList, IntPtr error) {
```

Any errors are ignored, those should be returned somehow.